src/sardana/spock/parser.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+
+##############################################################################
+##
+## This file is part of Sardana
+##
+## http://www.sardana-controls.org/
+##
+## Copyright 2011 CELLS / ALBA Synchrotron, Bellaterra, Spain
+##
+## Sardana is free software: you can redistribute it and/or modify
+## it under the terms of the GNU Lesser General Public License as published by
+## the Free Software Foundation, either version 3 of the License, or
+## (at your option) any later version.
+##
+## Sardana is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU Lesser General Public License for more details.
+##
+## You should have received a copy of the GNU Lesser General Public License
+## along with Sardana.  If not, see <http://www.gnu.org/licenses/>.
+##
+##############################################################################
+
+"""This package contains macro parameters parsing utilities."""
+
+__all__ = ["ParamParser"]
+
+import re
+import collections
+
+# Token specification
+PARAM = r"(?P<PARAM>[^\[\]\s]+)"
+QUOTEDPARAM = r"\"(?P<QUOTEDPARAM>.*?)\""
+SINGQUOTEDPARAM = r"'(?P<SINGQUOTEDPARAM>.*?)'"
```

Comment:
This regexp when used with `re.findall` correctly strips quotes from the group. But when used with the scanner's `match` it does not. See this example:

~~~
    SINGQUOTEDPARAM = r"'(?P<SINGQUOTEDPARAM>.*?)'"
    text = "'2 3'"
    master_pat = re.compile(SINGQUOTEDPARAM)
    assert master_pat.findall(text) == ["2 3"]
~~~

Any reqexp expect have an idea why?